### ISSUE TITLE
Issue with redis-oplog + monti

### ISSUE THREAD

**Issue by skeetmtp** [monti-apm/monti-apm-agent#125]
We got reactivity issues when using monti apm version 2.49.2 + latest redis-oplog (curlofcode)

It seems related to "limit" option we use in our publish, when i remove the limit option there is no more issue.

I think I succeed to reduce to a simple test case by tweeking one of existing unit test : in fact, just adding {limit: 100}
I see no reason for this test to pass without limit, but fail with large limit

```
addTestWithRoundedTime.only('Database - Redis Oplog - Removed', function (test) {
  // limit publish
  const pub = RegisterPublication(() => TestData.find({}, {limit: 100}));

  TestData.remove({});

  const client = GetMeteorClient();
  const sub = SubscribeAndWait(client, pub);

  TestData.insert({ name: 'test1' });
  TestData.insert({ name: 'test2' });
  TestData.insert({ name: 'test3' });

  Meteor._sleepForMs(100);

  TestData.remove({ name: 'test2' });

  Meteor._sleepForMs(100);

  let metrics = FindMetricsForPub(pub);

  test.equal(metrics.totalObservers, 1);
  test.equal(metrics.liveRemovedDocuments, 1);

  TestData.remove({});

  Meteor._sleepForMs(100);

  metrics = FindMetricsForPub(pub);

  test.equal(metrics.totalObservers, 1);
  test.equal(metrics.liveRemovedDocuments, 3);

  sub.stop();
  TestData.remove({});

  Meteor._sleepForMs(100);
});

```

mtest --package ./ --once --release 2.14

```
[[[[[ Tests ]]]]]

=> Started proxy.
=> Started MongoDB.
=> Linted your app. No linting errors.
I20240620-18:14:04.025(2)? test-in-console listening
=> Started your app.

=> App running at: http://localhost:10780/
Running tests...
S: tinytest - Database - Redis Oplog - Removed : !!!!!!!!! FAIL !!!!!!!!!!!
{"groupPath":["tinytest","Database","Redis Oplog"],"test":"Removed","events":[{"sequence":1,"type":"fail","details":{"type":"assert_equal","expected":"1","actual":"0","not":false}}],"server":true}
```

**Comment by skeetmtp** [comment]
May be @renanccastro you have a clue  ? As I saw a commit from you related to this  https://github.com/l3mpire/monti-apm-agent/commit/3f2ad5755a134c144f733a65ef73649160eb7351

**Comment by zodern** [comment]
Thanks @skeetmtp. The reproduction was very helpful. It's fixed in v2.49.3.
